scripts/west_commands/run_common.py

```diff
@@ -128,16 +128,14 @@ def _build_dir(args, die_if_none=True):
     if args.build_dir:
         return args.build_dir
 
-    cwd = getcwd()
-    default = path.join(cwd, DEFAULT_BUILD_DIR)
-    if is_zephyr_build(default):
-        return default
-    elif is_zephyr_build(cwd):
-        return cwd
+    dir = find_build_dir(None)
```

Comment:
```
This actually changes the behavoir slightly, since the current working
directory (cwd) will now be checked before the default build folder
('build'). Since builds within builds are not supported this should not
be an issue.
```

Thanks for calling attention to that. This is actually an issue -- it breaks compatibility with documented behavior:

```
  -d BUILD_DIR, --build-dir BUILD_DIR
                        Build directory to obtain runner information from. If
                        not given, this command tries to use build/ and then
                        the current working directory, in that order.
```

I think the helper in zephyr should be considered buggy and search in the order we use for --build-dir documentation.

Reply:
>I think the helper in zephyr should be considered buggy and search in the order we use for --build-dir documentation.

Not sure if I understand you here, you mean `find_build_folder` should be considered buggy and the runners should instead use the 1.`build` 2.`cwd` order?

Reply:
Yep, as discussed on slack.